content/tutorials/pusher-adaptor.textile

```diff
@@ -0,0 +1,206 @@
+
+blang[ruby].
+  Here's some code that subscribes to a channel from the Ably ruby client library, then publishes to it from the Pusher ruby client library:
+
+  ```[ruby]
+    # Subscribe to the 'public:some_channel' channel with the Ably client
+    ablyChannel = ably.channels.get('public:some_channel')
+    ablyChannel.subscribe do |message|
+      puts "Ably client received a message: #{message.name} - #{message.data.inspect}"
+    end
+
+    # Publish to the 'some_channel' channel with the Pusher client.
+    # This is a public channel, so the real channel name is 'public:some_channel'
+    pusher.trigger('some_channel', 'event_name', {some: 'data sent by the Pusher client library'})
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/fa24b73
+
+h2. Download tutorial source code
+
+blang[javascript].
+  The complete source code of a working example of Ably and Pusher client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pusher-adaptor-javascript.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pusher-adaptor-javascript
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.html@ and opening the page in your browser.
+
+blang[nodejs].
+  The complete source code of a working example of Ably and Pusher client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pusher-adaptor-nodejs.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pusher-adaptor-nodejs
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.js@ and running the demo @node example.js@
+
+blang[ruby].
+  The complete source code of a working example of Ably and Pusher client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pusher-adaptor-ruby.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pusher-adaptor-ruby
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.rb@ and running the demo with @bundle exec ruby example.rb@
+
+h2. Next steps
+
+# Full usage notes for the Pusher adaptor are available at: "Using the Ably Pusher Protocol Adaptor":https://support.ably.io/support/solutions/articles/3000055086-using-the-ably-pusher-protocol-adaptor/.
```

Comment:
Can we add the first bullet "[Review the many reasons to migrate from Pusher to Ably](https://www.ably.io/compare/ably-vs-pusher)"